### PR TITLE
Dark mode implemetation

### PR DESCRIPTION
**Pull request by doanguyen**
We used CSS variables to implement the dark mode feature

image

### CODE REVIEW

File: app/dashboard/templates/dashboard/domain_detail/dns.html [hunk 1]
```diff
@@ -28,15 +28,15 @@ <h1 class="h3"> {{ custom_domain.domain }} </h1>
       </div>
 
       <div class="mb-2">Add the following MX DNS record to your domain. <br>
-        Please note that there's a point (<em>.</em>) at the end target addresses. <br>
-        Also some domain registrars (Namecheap, CloudFlare, etc) might use <em>@</em> for the root domain.
+        Please note that there's a point (<em class="text-default">.</em>) at the end target addresses. <br>
```
Comment (nguyenkims) [inline app/dashboard/templates/dashboard/domain_detail/dns.html:31]: do you think there's a way to apply `.text-default` automatically to `<em>`?

File: app/dashboard/templates/dashboard/domain_detail/dns.html [hunk 2]
```diff
@@ -28,15 +28,15 @@ <h1 class="h3"> {{ custom_domain.domain }} </h1>
...
+        Also some domain registrars (Namecheap, CloudFlare, etc) might use <em class="text-default">@</em> for the root domain.
       </div>
 
       {% for priority, email_server in EMAIL_SERVERS_WITH_PRIORITY %}
-        <div class="mb-3 p-3" style="background-color: #eee">
-          Domain: <em>{{ custom_domain.domain }}</em> or <em>@</em> <br>
+        <div class="mb-3 p-3 bg-secondary">
```
Comment (nguyenkims) [inline app/dashboard/templates/dashboard/domain_detail/dns.html:36]: 👍

File: static/assets/js/core.js [hunk 3]
```diff
@@ -104,4 +104,18 @@ $(document).ready(function() {
       });
     });
   }
+
+  /**Dark mode controller */
+  var darkmode = localStorage.getItem('dark-mode');
```
Comment (nguyenkims) [inline static/assets/js/core.js:109]: `localStorage` doesn't work in private mode in some browsers (I learnt it the hard way :) ). I think we can use `local-storage-polyfill.js` here which falls back to `cookie` automatically, here's how it's currently used in the code https://github.com/simple-login/app/blob/70ce48cd792b85914147c96bc70babbc4083edac/server.py#L475

**Comment by nguyenkims** [comment]
@doanguyen the PR seems great overall! Just wonder is there a way to not replace the whole `dashboard.css` file and instead putting necessary change in another file? This file is replaced when `tabler` releases a new version actually.

I wonder maybe we can even have a `dashboard.dark.css` file that is a dark-built of `tabler` and switch to this one when user chooses the dark theme? This might require some changes in the back-end though to serve the `dashboard.dark.css` instead of `dashboard.css` according to user preference.

File: app/dashboard/templates/dashboard/domain_detail/dns.html (hunk 1 above)
Comment (doanguyen) [reply app/dashboard/templates/dashboard/domain_detail/dns.html:31]: Yep, we can apply the style to the em directive globally. I am going to update it

File: static/assets/js/core.js (hunk 3 above)
Comment (doanguyen) [reply static/assets/js/core.js:109]: I saw the local storage polyfill but didn't know that it has problems with private mode. I'm going to update it

**Comment by doanguyen** [comment]
The bad news is that I need to hook into the `dashboard.css` file to add this functionality (the steps that I took were: add some css variables for bright/dark modes, replace the  hard-coded colors from some elements with the new css variables). Maybe I can extract the new  feature to another css file for better understand/compatibility

**Comment by doanguyen** [comment]
Closed in favor of #177

**Pull request closed without merging**